code/engine/src/core/input.h

```diff
@@ -0,0 +1,230 @@
+#pragma once
+
+#include "defines.h"
+
+enum mouse
+{
+    MOUSE_LEFT    = 0x01,
+    MOUSE_MID     = 0x02,
+    MOUSE_RIGHT   = 0x03,
+
+// Currently up to 4 extra buttons
+// Most should only use 2
+    MOUSE_X1      = 0x04,
+    MOUSE_X2      = 0x05,
+    MOUSE_X3      = 0x06,
+    MOUSE_X4      = 0x07,
+
+    MAX_MOUSE_BUTTONS,
+};
+
+/*
+    Anything that can one-to-one with ascii follows ascii
+    Keypad on 0x6
+    Function on 0x7 and 0x8
+    Arrow Keys and Locks on 0xA
+    Extra Utility Keys on 0xB
+    IME keys on 0xF (for foreign [international?])
+*/
+enum keyboard
```

Comment:
Noteworthy: I added a bunch of keys that aren't supported in the platform layer... thoughts?